allensdk/brain_observatory/behavior/behavior_project_cache/project_cache_base.py

```diff
@@ -0,0 +1,292 @@
+from typing import Optional, Union, List
+from pathlib import Path
+import logging
+
+from allensdk.brain_observatory.behavior.behavior_project_cache.\
+    project_apis.data_io import ProjectCloudApiBase
+from allensdk.core.authentication import DbCredentials
+
+
+class ProjectCacheBase(object):
+
+    BUCKET_NAME: str = None
+    PROJECT_NAME: str = None
+
+    def __init__(
+            self,
+            fetch_api: ProjectCloudApiBase,
```

Comment:
Typing should also specify that this can accept a `BehaviorProjectLimsApi` (in case someone instantiates a `VisualBehaviorOphysProjectCache.from_lims()`)

Reply:
good point.